embiggen texty columns

Joel attempted a 0.19 -> 0.21 migration on a Firecloud dev database and hit an overflow migrating call outputs:

```
Oct 20 19:11:12 gce-cromwell-dev203 docker/cromwell-app[5085]: Caused by: liquibase.exception.MigrationFailedException: Migration failed for change set changesets/metadata_migration.xml::call_output_symbol_table_migration::tjeandet:
Oct 20 19:11:12 gce-cromwell-dev203 docker/cromwell-app[5085]:     Reason: liquibase.exception.UnexpectedLiquibaseException: liquibase.exception.CustomChangeException: Could not apply migration script for metadata at CallOutputSymbolTableMigration
.
.
.
Caused by: com.mysql.jdbc.MysqlDataTruncation: Data truncation: Data too long for column 'METADATA_VALUE' at row 1

```

This column is currently of type `TEXT` which has a size limit of 2^16 bytes.  Perusing the MySQL docs I see that `MEDIUMTEXT` supports 2^24 bytes and `LONGTEXT` supports 2^32 bytes, and these consume only 1 and 2 additional bytes over `TEXT` respectively.

Looking at our DB schema I'm also noticing that 

`CALL_CACHING_DETRITUS_ENTRY.DETRITUS_VALUE` has type `VARCHAR(255)`

which I could see ending badly with long GCS paths.

--- a/database/migration/src/main/resources/changelog.xml
+++ b/database/migration/src/main/resources/changelog.xml
@@ -45,6 +45,7 @@
     <include file="changesets/restart_and_recover_migration.xml" relativeToChangelogFile="true"/>
     <include file="changesets/summary_status_table.xml" relativeToChangelogFile="true"/>
     <include file="changesets/standardize_column_names.xml" relativeToChangelogFile="true" />
+    <include file="changesets/embiggen_metadata_value.xml" relativeToChangelogFile="true" />
     <include file="changesets/metadata_migration.xml" relativeToChangelogFile="true" />
     <include file="changesets/call_caching_job_detritus.xml" relativeToChangelogFile="true"/>
     <include file="changesets/job_store_tinyints.xml" relativeToChangelogFile="true" />

--- a/database/migration/src/main/resources/changesets/embiggen_metadata_value.xml
+++ b/database/migration/src/main/resources/changesets/embiggen_metadata_value.xml
@@ -1,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet author="mcovarr" id="entry_or_journal_existence_xor">
+        <preConditions>
+            <or>
+                <and>
+                    <tableExists tableName="METADATA_JOURNAL"/>
+                    <not>
+                        <tableExists tableName="METADATA_ENTRY"/>
+                    </not>
+                </and>
+                <and>
+                    <tableExists tableName="METADATA_ENTRY"/>
+                    <not>
+                        <tableExists tableName="METADATA_JOURNAL"/>
+                    </not>
+                </and>
+            </or>
+        </preConditions>
+        <comment>
+            Either METADATA_JOURNAL or METADATA_ENTRY must exist, but not both (Liquibase doesn't have an xor).
+        </comment>
+    </changeSet>
+
+    <changeSet author="mcovarr" id="embiggen_metadata_entry">
+        <preConditions onFail="MARK_RAN">
+            <tableExists tableName="METADATA_ENTRY"/>
+        </preConditions>
+        <comment>
+            The METADATA_ENTRY version of the embiggener. This should gracefully handle the absence of the table
+            in the event the table is still called METADATA_JOURNAL when this runs.
+        </comment>
+        <modifyDataType tableName="METADATA_ENTRY" columnName="METADATA_VALUE" newDataType="LONGTEXT"/>
+    </changeSet>
+
+    <changeSet author="mcovarr" id="embiggen_metadata_journal">
+        <preConditions onFail="MARK_RAN">
+            <tableExists tableName="METADATA_JOURNAL"/>
+        </preConditions>
+        <comment>
+            The METADATA_JOURNAL version of the embiggener. This should gracefully handle the absence of the table
+            in the event the table has been renamed to METADATA_ENTRY when this runs.
+        </comment>
+        <modifyDataType tableName="METADATA_JOURNAL" columnName="METADATA_VALUE" newDataType="LONGTEXT"/>
+    </changeSet>
+
+</databaseChangeLog>

